### DISCUSSION TITLE
throw render() within onBeforeRender hook on client side

### DISCUSSION THREAD

**Discussion by bjin7998** [vikejs/vike#1107] [Discussion]
I am using vite-ssr-plugin v0.4.410 and client side routing. The following code works fine on initial page load. However, when the user navigates to the specific route and the `throw render()` is triggered, the error page shows up without `is404` being set to `true`. 

```
11:42:25 AM [vite-plugin-ssr][request(6)] throw render(404) intercepted while rendering /de/64c22d8fa79b10a764d828cba/index.pageContext.json
11:42:25 AM [vite-plugin-ssr][request(6)] HTTP response /de/64c22d8fa79b10a764d828cba/index.pageContext.json 200
```

```
export default onBeforeRender

async function onBeforeRender (pageContext: PageContext) {
  try {
    const { id } = pageContext.routeParams.id
    const service = new Service()
    return {
      pageContext: {
        data:  {
          event: await service.getData(id)
        }
      }
    }
  } catch (e) {
    throw render(404)
  } 
}
```

**Comment by brillout** [comment]
Update to the latest version and try again.

**Reply by bjin7998** [reply]
Thanks for your fast response. Upgrading unfortunately does not solve it. `is404` is still `false`. And console print stays the same as well.

**Reply by brillout** [reply]
Is the only issue that `pageContext.is404` is `false` and the error page is correctly rendered, or is there anything else not working?

**Reply by brillout** [reply]
I believe 9dd96c31104e34f2cd9dc9313ccce0906a81dff1 fixes it. Pre-released as `0.4.140-commit-9dd96c3`.

**Reply by bjin7998** [reply]
Thanks a lot. Working as expected now. `is404` is `true` on client side's `throw render(404)`

**Reply by brillout** [reply]
👍 Btw. would your company be up for [sponsoring](https://github.com/sponsors/brillout)?
